doc/source/serve/advanced-guides/incremental-upgrade.md

```diff
@@ -0,0 +1,342 @@
+(rayservice-incremental-upgrade)=
```

Comment:
This doc should be under `doc/source/cluster/kubernetes/user-guides/` since `doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md` is there as well